scripts/edl_cmd_shell.py

```diff
@@ -284,8 +284,10 @@ def do_sdo_write(self, arg: str):
             value = float(args[3])
         elif obj.data_type == canopen.objectdictionary.VISIBLE_STRING:
             value = args[3]
+        elif obj.data_type == canopen.objectdictionary.DOMAIN:
+            value = args[3].encode("ascii")
```

Comment:
This assume the value is string, domain type are void, so this wont work in all cases. It would be more robust to expect the user to past a hex string for domain types.

Reply:
I guess the quick encode a string will work for now

Reply:
This is a good point, I would prefer to do it properly over quick hacks. Looking through CiA-301 and oresat-configs it feels like while officially undefined the intended use for `DOMAIN` is for file transfers; if the data is short enough to be written as a hex string easily it should generally be `OCTET_STRING`. What if instead for `DOMAIN` we expect a filename and `do_sdo_write()` opens and sends the contents?

On the other hand it looks like the type is wrong for `OS command` in oresat-configs. The `command` should be `OCTET_STRING` instead of `DOMAIN` (CiA-301 7.4.8.6). Additionally 7.5.2.26 says `command` should be "ISO8859 [ASCII] characters or binary". So given that, what if in `do_sdo_write()` for `OCTET_STRING` we try to decode the argument from a hex string to `bytes` and if that fails then just `.encode("ascii")` the given string? That way we cans still send shell commands directly through `sdo_write`.

In code:
```python
...
elif obj.data_type == canopen.objectdictionary.DOMAIN:
    with open(arg[3], 'rb') as f:
        value = f.read()
elif obj.data_type == canopen.objectdictionary.OCTET_STRING:
    try:
        value = bytes.fromhex(arg[3])
    except ValueError:
        value = arg[3].encode("ascii")
```

Reply:
Yea, that sounds like a good plan